### PR TITLE
Harmonize initialization between batch(scalar) and batch{scalar}

### LINKED ISSUE #775: single value batch initialisation failing for double on avx
Screenshot (133)

It initializes the first value with all other garbage. The broadcast is failing.
https://xsimd.readthedocs.io/en/latest/api/xsimd_batch.html#_CPPv4N5xsimd5batch5batchE1T

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Fix #775

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@marzer / @wermos : I'll happily take your review on that one!

File: include/xsimd/types/xsimd_batch.hpp [hunk 1]
```diff
@@ -451,14 +437,15 @@ namespace xsimd
...
      */
     template <class T, class A>
-    inline batch<T, A>::batch(std::initializer_list<T> data) noexcept
-        : batch(data.begin(), detail::make_index_sequence<size>())
+    template <class... Ts>
+    inline batch<T, A>::batch(T val0, T val1, Ts... vals) noexcept
+        : batch(kernel::set<A>(batch {}, A {}, val0, val1, static_cast<T>(vals)...))
     {
-        details::check_batch_init(*this, data);
+        static_assert(sizeof...(Ts) + 2 == size, "as many arguments as batch elements");
```
Comment (wermos) [inline include/xsimd/types/xsimd_batch.hpp:448]: Maybe the static assert message should say something like `"The constructor requires as many arguments as batch elements."`

**Review by wermos — COMMENTED** [review]
Everything else looks alright to me 👍

File: include/xsimd/types/xsimd_batch.hpp (hunk 1 above)
Comment (marzer) [reply include/xsimd/types/xsimd_batch.hpp:448]: Yup, this would be only note too. Looks great to me otherwise!

**Comment by serge-sans-paille** [comment]
`static_assert` updated, thanks @marzer and @wermos for the feedback and the great discussion we had on improving the batch constructor o/

**Pull request merged**